messenger.rst

```diff
@@ -612,26 +612,83 @@ Some third-party bundles and libraries provide configurable middleware via
 factories. Defining such requires a two-step configuration based on Symfony's
 :doc:`dependency injection </service_container>` features:
 
-.. code-block:: yaml
-
-    services:
-
-        # Step 1: a factory class is registered as a service with the required
-        # dependencies to instantiate a middleware
-        doctrine.orm.messenger.middleware_factory.transaction:
-            class: Symfony\Bridge\Doctrine\Messenger\DoctrineTransactionMiddlewareFactory
-            arguments: ['@doctrine']
-
-        # Step 2: an abstract definition that will call the factory with default
-        # arguments or the ones provided in the middleware config
-        messenger.middleware.doctrine_transaction_middleware:
-            class: Symfony\Bridge\Doctrine\Messenger\DoctrineTransactionMiddleware
-            factory: ['@doctrine.orm.messenger.middleware_factory.transaction', 'createMiddleware']
-            abstract: true
-            # the default arguments to use when none provided from config. Example:
-            # middleware:
-            #     - doctrine_transaction_middleware: ~
-            arguments: ['default']
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+
+            # Step 1: a factory class is registered as a service with the required
+            # dependencies to instantiate a middleware
+            doctrine.orm.messenger.middleware_factory.transaction:
```

Comment:
Unrelated to this but ... is there any reason for keep using snake-case IDs for services instead of FQCN IDs?

Reply:
Here, potentially there could be other service for this class too. If they all used the class name as the service id would cause conflicts. So I think it's better to use custom IDs for these services.